cmd/deployment/apm/create.go

```diff
@@ -18,59 +18,123 @@
+			if err != nil && err != cmdutil.ErrNodefinitionLoaded {
+				return err
+			}
 		}
-		defer f.Close()
 
-		var definition models.CreateApmRequest
-		if err := json.NewDecoder(f).Decode(&definition); err != nil {
-			return err
+		if payload == nil {
+			p, err := depresource.NewApm(depresource.NewStateless{
+				DeploymentID:       id,
+				ElasticsearchRefID: esRefID,
+				API:                ecctl.Get().API,
+				RefID:              refID,
+				Version:            version,
+				Region:             region,
+				TemplateID:         dt,
+				Size:               size,
+				ZoneCount:          zoneCount,
+			})
+			if err != nil {
+				return err
+			}
+			payload = p
 		}
 
-		track, _ := cmd.Flags().GetBool("track")
-		res, err := apm.Create(apm.CreateParams{
-			API:              ecctl.Get().API,
-			ID:               cmd.Flag("id").Value.String(),
-			CreateApmRequest: definition,
-			TrackParams: util.TrackParams{
-				Track:  track,
-				Output: ecctl.Get().Config.OutputDevice,
+		// Returns the ApmPayload skipping the creation of the resources.
+		if generatePayload {
+			return ecctl.Get().Formatter.Format("", payload)
+		}
+
+		var updateParams = deployment.UpdateParams{
+			DeploymentID: id,
+			API:          ecctl.Get().API,
+			Request: &models.DeploymentUpdateRequest{
+				// Setting PruneOrphans to false since we don't want any side
+				// effects on the deployment when only a partial deployment
+				// definition is sent.
+				PruneOrphans: ec.Bool(false),
+				Name:         name,
+				Resources: &models.DeploymentUpdateResources{
+					Apm: []*models.ApmPayload{payload},
+				},
 			},
-		})
+		}
+
+		res, err := deployment.Update(updateParams)
 		if err != nil {
 			return err
 		}
-		return ecctl.Get().Formatter.Format(filepath.Join("apm", "create"), res)
+
+		if err := ecctl.Get().Formatter.Format("", res); err != nil {
+			if !track {
+				return err
+			}
+			fmt.Fprintln(ecctl.Get().Config.OutputDevice, err)
+		}
+
+		if !track {
+			return nil
+		}
+
+		return depresource.TrackResources(depresource.TrackResourcesParams{
+			API:          ecctl.Get().API,
+			Resources:    res.Resources,
+			Orphaned:     res.ShutdownResources,
+			OutputDevice: ecctl.Get().Config.OutputDevice,
+		})
 	},
 }
 
 func init() {
-	createApmCmd.Flags().StringP("file", "f", "", "Deployment definition")
-	createApmCmd.Flags().String("id", "", "Overrides the deployment ID on which it's tied")
-	createApmCmd.Flags().Bool("track", false, cmdutil.TrackFlagMessage)
+	Command.AddCommand(createApmCmd)
+	createApmCmd.Flags().BoolP("track", "t", false, cmdutil.TrackFlagMessage)
+	createApmCmd.Flags().StringP("file", "f", "", "ApmPayload file definition. See help for more information")
+	createApmCmd.Flags().String("deployment-template", "", "Optional deployment template ID, automatically obtained from the current deployment")
+	createApmCmd.Flags().String("version", "", "Version to use, if not specified, the deployment's stack version will be used")
```

Comment:
```suggestion
	createApmCmd.Flags().String("version", "", "Optional version to use. If not specified, it will default to the deployment's stack version")
```